uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar.component.js

```diff
@@ -0,0 +1,351 @@
+     * @return {Promise}
+     */
+    function loadFiltersOptionValues () {
+      return OptionGroup.valuesOf([
+        'hrjc_region',
+        'hrjc_location',
+        'hrjc_level_type',
+        'hrjc_department'
+      ])
+      .then(function (data) {
+        vm.filters.optionValues.regions = data.hrjc_region;
+        vm.filters.optionValues.locations = data.hrjc_location;
+        vm.filters.optionValues.levelTypes = data.hrjc_level_type;
+        vm.filters.optionValues.departments = data.hrjc_department;
+      });
+    }
+
+    /**
+     * Loads all the public holidays
+     *
+     * @return {Promise}
+     */
+    function loadPublicHolidays () {
+      return PublicHoliday.all();
+    }
+
+    /**
+     * Loads all the data needed for the child components
+     *
+     * @return {Promise}
+     */
+    function loadSupportData () {
+      return $q.all([
+        loadAbsenceTypes(),
+        loadPublicHolidays(),
+        loadBasicOptionValues()
+      ])
+      .then(function (results) {
+        vm.supportData.absenceTypes = results[0];
+        vm.supportData.publicHolidays = results[1];
+        vm.supportData.leaveRequestStatuses = results[2].hrleaveandabsences_leave_request_status;
+        vm.supportData.dayTypes = results[2].hrleaveandabsences_leave_request_day_type;
+      });
+    }
+
+    /**
+     * Returns the structure of the month of the given date
+     *
+     * @param  {Object} dateMoment
+     * @return {Object}
+     */
+    function monthStructure (dateMoment) {
+      return {
+        index: dateMoment.month(),
+        year: dateMoment.year(),
+        name: dateMoment.format('MMM')
+      };
+    }
+
+    /**
+     * Reloads the selected months data
+     *
+     * If the source of the refresh is a period change, then
+     * it rebuilds the months list as well
+     * If the source of the refresh is a change in contacts filters, then
+     * it reloads the contacts as well
+     *
+     * @param {string} source The source of the refresh (period or contacts change)
+     */
+    function refresh (source) {
+      source = _.includes(['contacts', 'period'], source) ? source : 'period';
+
+      $q.resolve()
+        .then((source === 'period' ? buildPeriodMonthsList : _.noop))
+        .then((source === 'contacts' ? loadContacts : _.noop))
+        .then(function () {
+          injectAndShowMonths((source === 'contacts'));
+        });
+    }
+
+    /**
+     * Sends the "show" signal to the leave-calendar-month components
+     *
+     * @param {Boolean} forceDataReload if true, then a month will load its data
+     *   regardless if it had already loaded it
+     */
+    function sendShowMonthsSignal (forceDataReload) {
+      var monthsToShow = !vm.selectedMonths.length
+        ? vm.months
+        : vm.months.filter(function (month) {
+          return _.includes(vm.selectedMonths, month.index);
+        });
+
+      $rootScope.$emit('LeaveCalendar::showMonths', monthsToShow, !!forceDataReload);
```

Comment:
That's the point, it's to cater in case the function is mis-called with the wrong parameters. I don't think that making sure that the param is passed boolean is a big deal